report/renderer/components.js

```diff
@@ -0,0 +1,875 @@
+'use strict';
+
+// auto-generated by lighthouse-core/scripts/make-component.js
+
+/** @typedef {import('./dom.js').DOM} DOM */
+
+/* eslint-disable max-len */
+
+
+/**
+ * @param {DOM} dom
+ */
+function create3pFilterComponent(dom) {
+  const v0 = dom.document().createDocumentFragment();
+  const v1 = dom.createElement('style');
+  v1.append(dom.document().createTextNode('\n    .lh-3p-filter {\n      background-color: var(--table-higlight-background-color);\n      color: var(--color-gray-600);\n      float: right;\n      padding: 6px;\n    }\n    .lh-3p-filter-label, .lh-3p-filter-input {\n      vertical-align: middle;\n      user-select: none;\n    }\n    .lh-3p-filter-input:disabled + .lh-3p-ui-string {\n      text-decoration: line-through;\n    }\n  '));
+  v0.append(v1);
+  const v2 = dom.createElement('div', 'lh-3p-filter');
+  const v3 = dom.createElement('label', 'lh-3p-filter-label');
+  const v4 = dom.createElement('input', 'lh-3p-filter-input');
+  v4.setAttribute('type', 'checkbox');
+  v4.setAttribute('checked', '');
```

Comment:
might not be worth code golfing on this (straightforward generation code is probably the biggest win and it's nicely simple right now), but I wonder if some combo of `element.append(...)` and a setAttributes helper (humorous given #12796) could make this slightly easier for humans to follow/debug